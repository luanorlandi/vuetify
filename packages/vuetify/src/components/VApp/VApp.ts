// Styles
import './VApp.sass'

// Components
import VThemeProvider from '../VThemeProvider'

// Utilities
import { defineComponent, h } from 'vue'
import makeProps from '@/util/makeProps'

export default defineComponent({
  name: 'VApp',

  props: makeProps({
    theme: String,
  }),

  setup (props, { slots }) {
    return () => (
      h(VThemeProvider, {
        ...props,
        class: 'v-application',
        'data-app': true,
      }, () => h('div', { class: 'v-application__wrap' }, slots.default?.()))
    )
  },
})
