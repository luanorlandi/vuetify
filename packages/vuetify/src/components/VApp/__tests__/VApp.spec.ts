// Components
import VApp from '../VApp'

// Utilities
import { mount } from '@vue/test-utils'
import { createMockVuetifyInstance } from '../../../../test'

describe('VApp', () => {
  it('should match a snapshot', () => {
    const wrapper = mount(VApp, {
      global: {
        provide: {
          ...createMockVuetifyInstance(),
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have data-app attribute', () => {
    const wrapper = mount(VApp, {
      global: {
        provide: {
          ...createMockVuetifyInstance(),
        },
      },
    })

    const app = wrapper.find('.v-application')
    expect(app.attributes()['data-app']).toBe('true')
  })

  it('should allow a custom id', () => {
    const wrapper = mount(VApp, {
      attrs: {
        id: 'inspire',
      },
      global: {
        provide: {
          ...createMockVuetifyInstance(),
        },
      },
    })
    const app = wrapper.find('.v-application')
    expect(app.attributes().id).toBe('inspire')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
