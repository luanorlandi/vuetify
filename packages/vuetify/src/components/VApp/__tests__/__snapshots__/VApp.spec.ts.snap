// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VApp should allow a custom id 1`] = `
<div class="v-theme-provider theme--light v-application"
     data-app="true"
     id="inspire"
>
  <div>
    <div class="v-application__wrap">
    </div>
  </div>
</div>
`;

exports[`VApp should match a snapshot 1`] = `
<div class="v-theme-provider theme--light v-application"
     data-app="true"
>
  <div>
    <div class="v-application__wrap">
    </div>
  </div>
</div>
`;
